app/internal/import_holidays.py

```diff
@@ -0,0 +1,64 @@
+import re
+from datetime import datetime, timedelta
+
+from app.database.models import User, Event, UserEvent
+from sqlalchemy.orm import Session
+from typing import List, Match
+
+REGEX_EXTRACT_HOLIDAYS = re.compile(
+    r'SUMMARY:(?P<title>.*)(\n.*){1,8}DTSTAMP:(?P<date>\w{8})',
+    re.MULTILINE)
+
+
+def get_holidays_from_file(file: List[Event], session: Session) -> List[Event]:
+    """
+    This function using regex to extract holiday title
+    and date from standrd ics file
+    :param file:standard ics file
+    :param session:current connection
+    :return:list of holidays events
+    """
+    parsed_holidays = REGEX_EXTRACT_HOLIDAYS.finditer(file)
+    holidays = []
+    for holiday in parsed_holidays:
+        holiday_event = create_holiday_event(
+            holiday, session.query(User).filter_by(id=1).first().id)
+        holidays.append(holiday_event)
```

Comment:
Maybe it's better to return them as an iterator? (`yield`)

Reply:
Im saving the list all together, so iterator is not the right case here IMO.